src/Modules/LiveRankingModule/Templates/LiveRanking.mt

```diff
@@ -1,102 +1,73 @@
+    <import component="LiveRankingModule.Components.RecordRow" as="RecordRow"/>
 
-    <property type="List<LiveRankingWidgetPosition>" name="rankings"/>
-
-    <property type="double" name="scale" default="0.9"/>
-    <property type="double" name="w" default="68.0"/>
-    <property type="double" name="x" default="-160.0"/>
-    <property type="double" name="y" default="78.0"/>
-    <property type="double" name="headerHeight" default="8.0"/>
-    <property type="double" name="rowHeight" default="6.0"/>
-    <property type="double" name="rowSpacing" default="1.0"/>
-    <property type="int" name="rowsVisible" default="4"/>
-    
-    <property type="string" name="headerColor" default="c21d62"/>
-    <property type="string" name="primaryColor" default="4357ea"/>
-    <property type="string" name="playerRowBackgroundColor" default="999999"/>
-    <property type="string" name="logoUrl" default=""/>
+    <property type="ILiveRankingSettings" name="settings"/>
+    <property type="IEnumerable<LiveRankingPosition>" name="scores"/>
+    <property type="bool" name="isPointsBased"/>
 
     <template>
-        <frame id="live_rankings" pos="{{ x }} {{ y }}" size="{{ w }} 999" scale="{{ scale }}" z-index="100">
-            <frame>
-                <frame size="{{ w }} {{ headerHeight - 0.1 }}">
-                    <!-- HEADER -->
-                    <quad pos="{{ w * 0.5 - 10.0 }} 0.15"
-                          size="{{ headerHeight + 0.3 }} {{ w + 20.1 }}"
-                          valign="center"
-                          style="UICommon64_1"
-                          substyle="BgFrame1"
-                          colorize="{{ Theme.LiveRankingModule_LiveRanking_Default_BgHeaderGrad1 }}"
-                          rot="90"
-                    />
+        <UIStyle/>
+        <stylesheet>
+            <style
+                    class="lr-body-primary"
+                    bgcolor="{{ Theme.UI_LiveRankingModule_Widget_RowBg }}"
+                    opacity="0.9"
+            />
 
-                    <!-- GRADIENT -->
-                    <quad size="{{ w }} {{ headerHeight + 0.1 }}"
-                          image="file://Media/Painter/Stencils/15-Stripes/_Stripe0Grad/Brush.tga"
-                          modulatecolor="{{ Theme.LiveRankingModule_LiveRanking_Default_BgHeaderGrad2 }}"
-                    />
-
-                    <!-- LABEL -->
-                    <label pos="2 {{ headerHeight / -2.0 - 0.4 }}"
-                           text="Live Ranking"
-                           valign="center2"
-                           textfont="{{ Font.Bold }}"
-                           textcolor="{{ Theme.LiveRankingModule_LiveRanking_Default_Text }}"
-                           textprefix="$i$t"
-                           textsize="2"
-                    />
+            <style
+                    class="lr-body-highlight"
+                    bgcolor="{{ Theme.UI_LiveRankingModule_Widget_RowBgHighlight }}"
```

Comment:
is both the class and setting the bgcolor needed here? Isn't the class just doing that already?

Reply:
Thats the definition of the class, I've moved it to a separate component.